isotovideo

```diff
@@ -242,6 +243,13 @@ $bmwqemu::vars{TEST_GIT_HASH} = checkout_git_refspec($bmwqemu::vars{CASEDIR} =>
 # set a default distribution if the tests don't have one
 $testapi::distri = distribution->new;
 
+if (my $specfile = file('plugins.yaml')->slurp) {
+    my $spec = YAML::PP->new->load_file($specfile);
+    foreach my $plugin ($spec->{plugins}) {
+        checkout_git_repo_and_branch($plugin);
+    }
+}
+
```

Comment:
so maybe

```suggestion
checkout_git_repo_and_branch($_) for YAML::PP->new->load_file('plugins.yaml')->{plugins} if -e 'plugins.yaml';
```

Reply:
I suppose in a non-draft version we maybe want error handling here anyways, e.g. some meaningful error message if plugins doesn't exist or is no array would be useful for the user (instead of just some Perl warning).

Reply:
You mean if the file exists but is ill-formated?

Reply:
Yes. Maybe it can be validated against a schema (so we don't have to care about every error case ourselves).